src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionDescriptorBuilder.cs

```diff
@@ -137,96 +137,38 @@ public static IList<ControllerActionDescriptor> Build(ApplicationModel applicati
             ControllerModel controller,
             ActionModel action)
         {
-            var controllerAttributeRoutes = controller.Selectors
```

Comment:
I'm not entirely sure what this check was meant for. Essentially, this ensures at least one controller selector has an attribute route associated. If it is, the method combined it with all controller selectors. I would think we only want to merge it with controller selectors that have an associated attribute route. Doing so causes one of our unit tests to fail.

Reply:
Right now, I'm just ignoring this check - https://github.com/aspnet/Mvc/pull/6930/files#diff-f9ecc4fe4438037c1ffb1d25eac827f3R30

Reply:
If you're still unsure about this it's going to require some more digging. A concrete example to discuss would really help.

I would be super super super careful making any changes to this code.

Reply:
Fair enough. I'll leave the thing in and file a separate issue for this.